controllers/flinkcluster_util.go

```diff
@@ -566,9 +567,19 @@ func getFlinkJobSubmitLog(observedPod *corev1.Pod) (*FlinkJobSubmitLog, error) {
 		return nil, fmt.Errorf("no job pod found, even though submission completed")
 	}
 	var containerStatuses = observedPod.Status.ContainerStatuses
-	if len(containerStatuses) == 0 ||
-		containerStatuses[0].State.Terminated == nil ||
-		containerStatuses[0].State.Terminated.Message == "" {
+	if len(containerStatuses) == 0 {
+		return nil, fmt.Errorf("job pod found, but no termination log found even though submission completed")
+	}
+
+	var mainContainerStatus corev1.ContainerStatus
+	for i := 0; i < len(containerStatuses); i++ {
+		if containerStatuses[i].Name == "main" {
+			mainContainerStatus = containerStatuses[i]
+		}
+	}
+
+	if mainContainerStatus.State.Terminated == nil ||
```

Comment:
do we need a nil check on `mainContainerStatus`, in the case where it's not found in the above loop?